crayfish/animation.py

```diff
@@ -88,13 +93,12 @@ def animation(cfg, progress_fn=None):
         formattedTime = l.formatTime(time)
         # Set to render next timesteps
         rs = l.rendererSettings()
-        asd = rs.activeScalarDataset()
-        if asd.isValid():
-            rs.setActiveScalarDataset(QgsMeshDatasetIndex(asd.group(), i))
-        avd = rs.activeVectorDataset()
-        if avd.isValid():
-            rs.setActiveVectorDataset(QgsMeshDatasetIndex(avd.group(), i))
-        l.setRendererSettings(rs)
+        asg = rs.activeScalarDatasetGroup()
+        if asg >= 0:
+            l.setStaticScalarDatasetIndex(QgsMeshDatasetIndex(asg, i))
+        avg = rs.activeVectorDatasetGroup()
+        if avg >= 0:
+            l.setStaticVectorDatasetIndex(QgsMeshDatasetIndex(avg, i))
```

Comment:
Wouldn't be possible to change the time with the new temporal framework and not through the `l.setStaticScalarDatasetIndex`? 
Preferably I would just check if the mesh layer is temporal, and if not, just do nothing. If it is temporal, we have all the times, so we can just update temporal controller time for animation? The benefit would be that it will update all the time-dependent layers (e.g. WMS-T in the background), not just the mesh...